pkg/processor/trigger/http/trigger.go

```diff
@@ -433,6 +417,32 @@ func (h *http) handleActualRequest(ctx *fasthttp.RequestCtx) {
 	}
 }
 
+func (h *http) preflightRequestValidation(ctx *fasthttp.RequestCtx, origin string) bool {
+
+	// ensure origin is given, otherwise the request is outside the scope of CORS specifications
+	if !h.configuration.CORS.OriginAllowed(origin) {
+		return false
+	}
+
+	method := common.ByteSliceToString(ctx.Request.Header.Peek("Access-Control-Request-Method"))
+
+	// ensure method is given, otherwise the request is outside the scope of CORS specifications
+	if !h.configuration.CORS.MethodAllowed(method) {
+		return false
+	}
+
+	headers := common.ByteSliceToString(ctx.Request.Header.Peek("Access-Control-Request-Headers"))
+	if headers != "" {
+
+		// ensure request headers allowed (it can also be empty)
+		if !h.configuration.CORS.HeadersAllowed(strings.Split(headers, ", ")) {
+			h.UpdateStatistics(false)
```

Comment:
don't you update the stats twice here?

Reply:
Yup, fixed + added tests to ensure it gets update only once